engines/onboarding/app/controllers/onboarding/api/v1/invitations_controller.rb

```diff
@@ -2,61 +2,106 @@ module Onboarding
   module Api
     module V1
       class InvitationsController < ApplicationController
+        before_action :authenticate, only: [:bulk]
+        before_action :load_group, only: [:bulk]
+        before_action :load_invitation, only: [:show, :accept]
 
-        before_action :authenticate
-        before_action :load_group, only: [:create]
-        before_action :load_invited_user, only: [:create]
+        # Creates invitations in bulk
+        #
+        # POST /api/v1/invitations/bulk
+        #
+        def bulk
+          authorize @group
+
+          emails = bulk_params.fetch(:emails, '')
+
+          if emails.empty?
+            return render(
+              status: :bad_request,
+              json: {errors: {emails: t('onboarding.invitation.bulk.errors.empty') }}
+            )
+          end
+
+          emails = emails.split(',')
+          valids = valid_emails(emails)
+
+          if valids.any?
+            # We pick first 100 valid emails
+            # We don't want to allow infinite invitations.
+            valid_emails = valids.first(100)
+            InvitationService.new.bulk_invite!(@group, current_user, valid_emails)
```

Comment:
all logic from line 26 until here should not be in the controller. Please, move it to InvitationService, and if needed (very likely) move those responsibilities to a new object. This way, it will be easier to reuse and the controller will conform to the single responsibility principle.